### PR TITLE
Add sandbox mode

### PR DESCRIPTION
**Pull request by wietsedecnijf**

### CODE REVIEW

File: src/MailjetAdapter.php [hunk 1]
```diff
@@ -33,6 +33,21 @@ public static function displayName(): string
...
+
+    public function __construct($config = [])
+    {
+        // Config normalization
+        if (($config['useSandboxMode'] ?? null) === '') {
+            unset($config['useSandboxMode']);
+        }
+
+        parent::__construct($config);
+    }
```
Comment (boboldehampsink) [inline src/MailjetAdapter.php:52]: Do you think this is necessary?

File: src/templates/settings.html [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
+errors: adapter.getErrors('apiSecret')
+}) }}
+
+{{ forms.booleanMenuField({
+label: "Use Sandbox Mode"|t('mailjet-adapter'),
+id: 'useSandboxMode',
+name: 'useSandboxMode',
+includeEnvVars: true,
+value: adapter.useSandboxMode
 }) }}
```
Comment (boboldehampsink) [inline src/templates/settings.html:31]: Can you fix the indenting of this file?

**Review by boboldehampsink — CHANGES_REQUESTED** [review]

**Comment by boboldehampsink** [comment]
Thanks @wietsedecnijf!

File: src/templates/settings.html (hunk 2 above)
Comment (wietsedecnijf) [reply src/templates/settings.html:31]: Done

File: src/MailjetAdapter.php (hunk 1 above)
Comment (wietsedecnijf) [reply src/MailjetAdapter.php:52]: I'm not sure, I took Brandon's implementation of the `useAuthentication` setting in the Smtp Adapter as an example: https://github.com/craftcms/cms/blob/72122c099e702529756d2f0a3bc5a582d632cd90/src/mail/transportadapters/Smtp.php#L71
I can remove it is you want.

Comment (boboldehampsink) [reply src/MailjetAdapter.php:52]: Must be good then ;-)

**Review by boboldehampsink — APPROVED** [review]

**Pull request merged**